pkg/controller/metrics/metricscontroller.go

```diff
@@ -0,0 +1,264 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS metrics controller service.
+
+*/
+
+package metrics
+
+import (
+	"encoding/json"
+	log "github.com/golang/glog"
+	"github.com/opensds/opensds/pkg/dock/client"
+	"github.com/opensds/opensds/pkg/model"
+	pb "github.com/opensds/opensds/pkg/model/proto"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+// Controller is an interface for exposing some operations of metric controllers.
+type Controller interface {
+	GetLatestMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricSpec, error)
+	GetInstantMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricSpec, error)
+	GetRangeMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricSpec, error)
+	SetDock(dockInfo *model.DockSpec)
+}
+
+// NewController method creates a controller structure and expose its pointer.
+func NewController() Controller {
+	return &controller{
+		Client: client.NewClient(),
+	}
+}
+
+type controller struct {
+	client.Client
+	DockInfo *model.DockSpec
+}
+
+// latest+instant metrics structs begin
+type InstantMetricReponseFromPrometheus struct {
+	Status string `json:"status"`
+	Data   Data   `json:"data"`
+}
+type Metric struct {
+	Name     string `json:"__name__"`
+	Device   string `json:"device"`
+	Instance string `json:"instance"`
+	Job      string `json:"job"`
+}
+type Result struct {
+	Metric Metric        `json:"metric"`
+	Value  []interface{} `json:"value"`
+}
+type Data struct {
+	ResultType string   `json:"resultType"`
+	Result     []Result `json:"result"`
+}
+
+// latest+instant metrics structs end
+
+// latest+instant metrics structs begin
+type RangeMetricReponseFromPrometheus struct {
+	Status string    `json:"status"`
+	Data   RangeData `json:"data"`
+}
+type RangeMetric struct {
+	Name     string `json:"__name__"`
+	Device   string `json:"device"`
+	Instance string `json:"instance"`
+	Job      string `json:"job"`
+}
+type RangeResult struct {
+	Metric RangeMetric     `json:"metric"`
+	Values [][]interface{} `json:"values"`
+}
+type RangeData struct {
+	ResultType string        `json:"resultType"`
+	Result     []RangeResult `json:"result"`
+}
+
+// latest+instant metrics structs end
+
+func (c *controller) GetLatestMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricSpec, error) {
+
+	var metrics []model.MetricSpec
+	// make a call to Prometheus, convert the response to our format, return
+	response, err := http.Get("http://localhost:9090/api/v1/query?query=" + opt.MetricName)
```

Comment:
Please add it as config. Remove hard-coding.

Reply:
https://github.com/opensds/opensds/issues/691